### DISCUSSION TITLE
Service for changing friendly_name of an entity (rename entity)

### DISCUSSION THREAD

**Discussion by FragMenthor** [frenck/spook#573] [Ideas] [answered]
First of all, thanks for all this work :)
If it's possible, I would like to request a service to change the friendly_name of an entity. That could be quite handy.
For instance, I have an input_boolean to switch my “Christmas Mode” on, when I assemble my tree, and I get some devices I have in storage for the lights; I keep track of how many devices I have offline, and when christmas mode is off I add a “®” to the start of the name, so that my templates ignore these devices as they are in “Reserve”; they are not considered to be offline. It would be nice to have an automation to do this when I turn on / off my christmas mode.
Other use case: For instance, a smart plug with power monitoring could change the name according to device consumption (if you know the consuption of the devices you connect there it is easy enough to ‘guess’ whats connected and change the name accordingly).
Any thoughts on this? Thanks in advance!

**Comment by teskanoo** [comment]
I think this covers the requested functionality: https://spook.boo/entities#rename-an-entity

**Comment by frenck** [comment, marked as answer]
This one is done. It shipped in Spook 4.0.0 as the `homeassistant.rename_entity` action, which sets the friendly name of an entity.

So your Christmas mode automation works: call it with the `®` prefix when you put the devices in storage, call it again without when the tree goes up.

Docs: https://spook.boo/entities#rename-an-entity

Sorry it took a while to come back and say so.

../Frenck

<p>
  <a href="https://www.linkedin.com/in/frenck/"><img src="https://github.com/frenck/frenck/raw/main/images/linkedin.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://youtube.com/@frenck"><img src="https://github.com/frenck/frenck/raw/main/images/youtube.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://bsky.app/profile/frenck.social"><img src="https://github.com/frenck/frenck/raw/main/images/bluesky.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://fosstodon.org/@frenck"><img src="https://github.com/frenck/frenck/raw/main/images/mastodon.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://www.instagram.com/frenck/"><img src="https://github.com/frenck/frenck/raw/main/images/instagram.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://www.threads.net/@frenck"><img src="https://github.com/frenck/frenck/raw/main/images/threads.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://www.facebook.com/frenck.dev/"><img src="https://github.com/frenck/frenck/raw/main/images/facebook.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://x.com/frenck"><img src="https://github.com/frenck/frenck/raw/main/images/x.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://www.tiktok.com/@frenck.nl"><img src="https://github.com/frenck/frenck/raw/main/images/tiktok.svg" width="18" height="18"></a>
</p>

Blogging my personal ramblings at <a href="https://frenck.dev">frenck.dev</a>
